### PR TITLE
Multiple options multiple select questions: There is no way to 'select nothing' #2511

### LINKED ISSUE #2511: Multiple options multiple select questions: There is no way to 'select nothing'
If nothing is selected, the response is not saved at all :-p

### PR DESCRIPTION
**Pull request by kanghj**
Fixes #2511

### CODE REVIEW

**Comment by kanghj** [comment]
This shows how the results page will look like. If the user selected "None of the above", the Feedback column will be blank. 
msqresponses
msqcsv

On the submission page, "None of the above" will appear in italics. On selecting "None of the above", other options will reset. If the user selects something else, then "None of the above" will be deselected.
submitpage

File: src/main/java/teammates/common/datatransfer/FeedbackQuestionDetails.java [hunk 1]
```diff
@@ -201,4 +201,20 @@ public String getNoResponseText(String giverEmail, String recipientEmail,
...
         return Const.INSTRUCTOR_FEEDBACK_RESULTS_MISSING_RESPONSE;
     }
+    
+    /**
+     * Checks if the question has been skipped. 
+     */
+    public boolean isQuestionSkipped(String[] answer) {
+        
+        boolean allAnswersEmpty = true;
+        if(answer!=null){
```
Comment (LowWeiLin) [inline src/main/java/teammates/common/datatransfer/FeedbackQuestionDetails.java]: Can do an early return if answer == null.

File: src/main/java/teammates/common/datatransfer/FeedbackResponseAttributes.java [hunk 2]
```diff
@@ -4,9 +4,11 @@
 import java.util.Collections;
 import java.util.Comparator;
 import java.util.List;
+import java.util.Map;
```
Comment (LowWeiLin) [inline src/main/java/teammates/common/datatransfer/FeedbackResponseAttributes.java]: Are these changes required?

**Comment by LowWeiLin** [comment]
It seems abit odd to leave the feedback column blank if None of the above is selected. I think showing "None of the above" is better?
@damithc What should be shown in this case?

**Comment by damithc** [comment]
Technically, not selecting any options is equivalent to an empty cell.
Let's leave it empty for now. If it causes problems later, we can use
something like 'none selected'.

**Comment by kanghj** [comment]
Changes made

**Comment by damithc** [comment]
Another one waiting to merge.

**Comment by kanghj** [comment]
Oops, I was waiting for the toMerge label, didn't notice it until now. Will merge now.
